README.md

```diff
@@ -3,19 +3,238 @@
+
+## Want to contribute to OPA?
+
+* See [DEVELOPMENT.md](./docs/devel/DEVELOPMENT.md) to build and test OPA itself.
+* See [CONTRIBUTING.md](./CONTRIBUTING.md) to get started.
+
+## How does OPA work?
+
+OPA gives you a high-level declarative language to author and enforce policies
+across your stack. With OPA, you define rules that govern how your system
+should behave and then you offload policy decisions from your service to OPA.
+
+The examples below show how you can use OPA to enable policy-based control in
+any system, at any layer. For concrete examples of how to integrate OPA with
+existing systems like
+[Kubernetes](http://www.openpolicyagent.org/docs/kubernetes-admission-control.html),
+[Terraform](http://www.openpolicyagent.org/docs/terraform.html),
+[Docker](http://www.openpolicyagent.org/docs/docker-authorization.html),
+[SSH](http://www.openpolicyagent.org/docs/ssh-and-sudo-authorization.html), and
+more, see [openpolicyagent.org](http://www.openpolicyagent.org).
+
+### Example: API Authorization
+
+This example shows how you can enforce access controls over salary information
+served by a simple HTTP API. In this example, users are allowed to access their
+own salary as well as the salary of anyone who reports to them.
+
+```ruby
+allow {
+    input.method = "GET"
+    input.path = ["salary", id]
+    input.user_id = id
+}
+
+allow {
+    input.method = "GET"
+    input.path = ["salary", id]
+    managers = data.management_chain[id]
+    id = managers[_]
+}
+```
+
+#### Example Queries
+
+**Is someone allowed to access their own salary?**
+
+```ruby
+> input = {"method": "GET", "path": ["salary", "bob"], "user_id": "bob"}
+> allow
+true
+```
+
+**Display the management chain for bob:**
+
+```ruby
+> data.management_chain["bob"]
+[
+    "ken",
+    "janet"
+]
+```
+
+**Is alice allowed to access bob's salary?**
+
+```ruby
+> input = {"method": "GET", "path": ["salary", "bob"], "user_id": "alice"}
+> allow
+false
+```
+
+**Is janet allowed to access bob's salary?**
+
+```ruby
+> input = {"method": "GET", "path": ["salary", "alice"], "user_id": "janet"}
+> allow
+true
+```
+
+### Example: App Placement
+
+This example shows how you can enforce where apps are deployed inside a simple
+orchestrator. In this example, apps must be deployed onto clusters that satisfy
+PCI and jurisdiction requirements.
+
+```ruby
+app_placement[cluster_id] {
+    cluster = data.clusters[cluster_id]
+    satisfies_jurisdiction(input.app, cluster)
+    satisfies_pci(input.app, cluster)
+}
+
+satisfies_jurisdiction(app, cluster) {
+    not app.tags["requires-eu"]
+}
+
+satisfies_jurisdiction(app, cluster) {
+    app.tags["requires-eu"]
+    startswith(cluster.region, "eu-")
+}
+
+satisfies_pci(app, cluster) {
+    not app.tags["requires-pci-level"]
+}
+
+satisfies_pci(app, cluster) {
+    level = to_number(app.tags["requires-pci-level"])
+    level >= cluster.tags["pci-level"]
+}
+```
+
+#### Example Queries
+
+**Display all clusters:**
+
```

Comment:
First query here should probably be the meatiest one.  (That comment applies to all the examples.)  In this case, I would expect the first query to be "which cluster(s) should I deploy this application to?".